Online/RPCs.cs

```diff
@@ -122,48 +122,81 @@ public static void GoToDeathScreen()
+            }
+
             foreach (OnlinePlayer player in OnlineManager.players)
             {
-                if (!player.OutgoingEvents.Any(e => e is RPCEvent rpc && rpc.IsIdentical(RPCs.GoToWinScreen, malnurished)))
+                if (!player.OutgoingEvents.Any(e => e is RPCEvent rpc && rpc.IsIdentical(RPCs.GoToWinScreen, malnourished, denPos)))
                 {
-                    player.InvokeRPC(RPCs.GoToWinScreen, malnurished);
+                    if (player.isMe)
+                    {
+                        GoToWinScreen(malnourished, denPos);
+                    }
+                    else
+                    {
+                        player.InvokeRPC(RPCs.GoToWinScreen, malnourished, denPos);
+                    }
                 }
             }
         }
 
         //Assumed to be called for storymode only
         [RPCMethod]
-        public static void GoToWinScreen(bool malnourished)
+        public static void GoToWinScreen(bool malnourished, string denPos)
         {
+            RainMeadow.Debug($"({malnourished})");
             var game = (RWCustom.Custom.rainWorld.processManager.currentMainLoop as RainWorldGame);
             if (game == null || game.manager.upcomingProcess != null)
             {
                 return;
             }
 
-            if (!malnourished && !game.rainWorld.saveBackedUp)
+            if (OnlineManager.lobby.isOwner)
             {
-                game.rainWorld.saveBackedUp = true;
-                game.rainWorld.progression.BackUpSave("_Backup");
+                if (!malnourished && !game.rainWorld.saveBackedUp)
+                {
+                    game.rainWorld.saveBackedUp = true;
+                    game.rainWorld.progression.BackUpSave("_Backup");
+                }
             }
-
-            //This needs to be called after shelterDoor::Close and game state synced for this to be accurate.
-            var denPos = (OnlineManager.lobby.gameMode as StoryGameMode).defaultDenPos;
-            if (denPos == null) {
-                AbstractCreature firstAlivePlayer = game.FirstAlivePlayer;
-                denPos = game.world.GetAbstractRoom(firstAlivePlayer.pos).name;
+            else
+            {
+                var storyClientSettings = OnlineManager.lobby.gameMode.clientSettings as StoryClientSettings;
+                if (storyClientSettings.isDead)
+                {
+                    storyClientSettings.myLastDenPos = denPos;
+                }
             }
 
-            //TODO: Having soft-win on makes this very difficult. For now I'm (Turtle) locking it down so that you can only win if the host is alive
-            if (OnlineManager.lobby.isOwner) {
-                game.GetStorySession.saveState.SessionEnded(game, true, malnourished);
-            }
+            // TODO: investigate client save desync (e.g. swallowed items)
+            game.GetStorySession.saveState.SessionEnded(game, true, malnourished);
```

Comment:
I'm not sure how this isn't crashing. Line 196 in Savestate. If a player is dead and unrealized, this should crash. and a dead player can be unrealized. 

PlayerSessionRecord needs to be synced as well.

Might need to just hook into sessionEnded and redo the logic to work with online versions of things

Reply:
can we just remove unrealized players from the player list? not sure how this works though

Reply:
add a null check to the realized player

Reply:
done in 19f7c0c1fd0ef9f64a18069bc2795d243dae9e38

Reply:
done in 19f7c0c1fd0ef9f64a18069bc2795d243dae9e38